### PR TITLE
Created "note" component

### PR DESCRIPTION
**Pull request by crookedspider1**

### CODE REVIEW

**Review by MD-Hamza — COMMENTED** [review]
Looks good

**Comment by dungwoong** [comment]
we can try merging this AFTER we merge playable_level to main maybe

File: PukingPredator/Assets/Scripts/UI/Note.cs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+    void OnTriggerEnter(Collider other)
+    {
+        if (isPlayerNearby) { return; }
+
+        if (other.gameObject.CompareTag(GameTag.player))
+        {
+            isPlayerNearby = true;
+            StopAllCoroutines(); // Stop any ongoing fading out
+            StartCoroutine(FadeImage(originalAlpha)); // Fade in to alpha = 1 * originalAlpha
```
Comment (dungwoong) [inline PukingPredator/Assets/Scripts/UI/Note.cs:49]: from my understanding, alpha is the original(max) alpha we want this image to achieve. Could we add a [SerializeField] to this to change it in the editor maybe?

File: PukingPredator/Assets/Scripts/UI/Note.cs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+    /// <summary>
+    /// Create the Image component dynamically.
+    /// </summary>
+    void CreateImageComponent()
+    {
+        var canvas = GameObject.Find("Canvas");
+
+        // Create a new GameObject and make it a child of the Canvas
+        GameObject imageObject = new GameObject("ProximityImage");
```
Comment (dungwoong) [inline PukingPredator/Assets/Scripts/UI/Note.cs:75]: if we have multiple notes, this would result in multiple "ProximityImage" gameobjects. Is that fine?

**Review by dungwoong — APPROVED** [review]
Looks good, left some small comments

File: PukingPredator/Assets/Scripts/UI/Note.cs (hunk 2 above)
Comment (crookedspider1) [reply PukingPredator/Assets/Scripts/UI/Note.cs:75]: Its not an issue, they each have their own reference to the exact image they are linked to, and its not that important to manually look at those objects via the unity inspector

File: PukingPredator/Assets/Scripts/UI/Note.cs (hunk 1 above)
Comment (crookedspider1) [reply PukingPredator/Assets/Scripts/UI/Note.cs:49]: there's not really much point, the images are intended to be fully opaque anyways. it frankly could have originalAlpha swapped to just 1 and have no real consequences in our use case

**Pull request merged**